.jenkins/pytorch/test.sh

```diff
@@ -51,6 +51,12 @@ else
   export PYTORCH_TEST_SKIP_NOARCH=1
 fi
 
+if [[ -n "$IN_PULL_REQUEST" ]]; then
```

Comment:
now that I think about it. this is problematic because there's no way for developer to override this via label and such..

Reply:
well there is if you introduce another env var FORCE_CUDA_MEM_LEAK_CHECK that can be controlled externally? And then you can change this condition to also include logic for the new env var

Reply:
even if we introduce it. there's no way to allow users to set this variable, is there? (e.g. if I add a ci/master label. `IN_PULL_REQUEST` is still `true` right?

Reply:
Is there another var we can base this off of? `IN_PULL_REQUEST` doesn't necessarily exist for GHA yet

Reply:
i think it is fine for now, as we dont have cuda tests on GHA on PR yet, correct?

Reply:
Apparently we got this for free: 
ci/master labeled jobs are launched in: https://app.circleci.com/pipelines/github/pytorch/pytorch?branch=pull%2F60116%2Fhead (including master_builds and normal builds)
--> these DO NOT have CIRCLE_PULL_REQUEST env var set

where as 
normal PR jobs are launched in https://app.circleci.com/pipelines/github/pytorch/pytorch?branch=pull%2F60116
--> these have CIRCLE_PULL_REQUEST env var set